### PR TITLE
EID-1887: Create DTO for Metatron response

### PR DESCRIPTION
**Pull request by Wynndow**
https://govukverify.atlassian.net/browse/EID-1887

Adds a DTO for use by Metatron and the client the apps will be using to fetch details from metadata for different countries.

### CODE REVIEW

**Review by JohnWattsGDS — DISMISSED** [review]

File: proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/MetatronResponse.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.net.URI;
+
+public class MetatronResponse {
+
+    private String samlSigningCertX509;
+    private String samlEncryptionCertX509;
+    private URI destination;
+    private String entityId;
+    private String countryCode;
```
Comment (idrop) [inline proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/MetatronResponse.java]: Should we annotate these fields with validators? We're doing it for other DTOs in this package. In this way we'd get a useful 400 error response in clients.

Comment (Wynndow) [reply proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/MetatronResponse.java]: I was wondering the same thing, but have had doubts for a while that we actually use these annotations properly elsewhere. If you look at the first commit of this PR I wrote some tests to check their behaviour. These validators only automatically come into play if you are using the DTO as the parameter type for a resource. If you're using it to deserialize a response (as we will be) they don't. _Unless_ you specifically new up and invoke a validator. It seems like the reasoning is to make it easy to validate stuff you're receiving from the outside world.

File: proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/MetatronResponse.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package uk.gov.ida.notification.contracts;
+
+import java.net.URI;
+
+public class MetatronResponse {
```
Comment (CharlesIC) [inline proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/MetatronResponse.java]: I know this is a cool name but maybe something like `CountryMetadataInfo` would be more descriptive of what this DTO represents?

Comment (Wynndow) [reply proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/MetatronResponse.java]: Perhaps, but it is in keeping with the naming conventions we've used for other DTO's in the proxy node, like `EidasSamlParserResponse` and `HubResponseTranslatorRequest`

Comment (CharlesIC) [reply proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/MetatronResponse.java]: I don't think those are particularly good names so would call this new DTO whatever is the best name, which I think is something specific and more descriptive

File: proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/MetatronResponse.java (hunk 1 above)
Comment (Wynndow) [reply proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/MetatronResponse.java]: Phil and I just dug into this. The validators can be used on the server side by annotating the resource with `@Valid`. If the object being serialized isn't valid a 500 is returned. I'm going to add validations and the `@Valid` annotation to this DTO.

**Review by CharlesIC — DISMISSED** [review]

**Review by JohnWattsGDS — DISMISSED** [review]
I'm pretty sure MetatronResponse is a superior name for this class but approving anyway ;)

**Review by CharlesIC — APPROVED** [review]

**Review by idrop — APPROVED** [review]

**Pull request merged**